packages/amplify-codegen/src/commands/add.js

```diff
@@ -20,72 +20,99 @@ const addWalkThrough = require('../walkthrough/add');
-        apiDetailSpinner.succeed();
-        await updateAmplifyMeta(context, apiDetails);
-        break;
-      } catch (e) {
-        apiDetailSpinner.fail();
-        if (e instanceof AmplifyCodeGenAPINotFoundError) {
-          context.print.info(`AppSync API was not found in region ${region}`);
-          ({ shouldRetry, region } = await changeAppSyncRegion(context, region));
-        } else {
-          throw e;
+  if (!context.withoutInit) {
+    if (!apiId) {
+      const availableAppSyncApis = getAppSyncAPIDetails(context); // published and un-published
+      if (availableAppSyncApis.length === 0) {
+        throw new NoAppSyncAPIAvailableError(constants.ERROR_CODEGEN_NO_API_AVAILABLE);
+      }
+      [apiDetails] = availableAppSyncApis;
+      apiDetails.isLocal = true;
+    } else {
+      let shouldRetry = true;
+      while (shouldRetry) {
+        const apiDetailSpinner = new Ora();
+        try {
+          apiDetailSpinner.start('Getting API details');
+          apiDetails = await getAppSyncAPIInfo(context, apiId, region);
+          apiDetailSpinner.succeed();
+          await updateAmplifyMeta(context, apiDetails);
+          break;
+        } catch (e) {
+          apiDetailSpinner.fail();
+          if (e instanceof AmplifyCodeGenAPINotFoundError) {
+            context.print.info(`AppSync API was not found in region ${region}`);
+            ({ shouldRetry, region } = await changeAppSyncRegion(context, region));
+          } else {
+            throw e;
+          }
         }
       }
     }
   }
 
-  if (!apiDetails) {
+  if (!context.withoutInit && !apiDetails) {
     return;
   }
   const answer = await addWalkThrough(context);
 
+
   let schema;
-  if (!apiDetails.isLocal) {
-    schema = await downloadIntrospectionSchemaWithProgress(
-      context,
-      apiDetails.id,
-      answer.schemaLocation,
-      region,
-    );
-  } else if (getFrontEndHandler(context) === 'android') {
-    schema = answer.schemaLocation;
+  if (!context.withoutInit) {
+    if (!apiDetails.isLocal) {
+      schema = await downloadIntrospectionSchemaWithProgress(
+        context,
+        apiDetails.id,
+        answer.schemaLocation,
+        region,
+      );
+    } else if (getFrontEndHandler(context) === 'android') {
+      schema = answer.schemaLocation;
+    } else {
+      schema = getSDLSchemaLocation(apiDetails.name);
+    }
   } else {
-    schema = getSDLSchemaLocation(apiDetails.name);
+    schema = './schema.json';
   }
 
-  const newProject = {
-    projectName: apiDetails.name,
-    includes: answer.includePattern,
-    excludes: answer.excludePattern,
-    schema,
-    amplifyExtension: {
-      codeGenTarget: answer.target || '',
-      generatedFileName: answer.generatedFileName || '',
-      docsFilePath: answer.docsFilePath,
-      region,
-    },
-  };
+  let newProject;
+  if (!context.withoutInit) {
+    newProject = {
+      projectName: apiDetails.name,
```

Comment:
Can we use ternary operator to switch to different name based on with or without init. Everything else looks the same in both the casses

Reply:
Changing to ternary operator

Reply:
Changing to ternary